File: examples/python/signature.py
from stellar_base.keypair import Keypair
import hashlib

def hash():
      mesg = "Some message"

      # Hash
      h = hashlib.sha256()
      h.update(mesg.encode())

      print(h.hexdigest())

def sign(hash):
      # GDMG5Z4XL3CNGHK2GJD5TFIDRWRCBFVFV3WAFWFSBONWB6AKDODILHFZ
      kp = Keypair.from_secret("SDWMABEXMMUVENWEB73FB3EQJB5QSKOYIBXDOXAE6A3NIHIYRUQJSWXY")

      # Sign
      signature = kp.signing_key.sign(hash, encoding='hex')

      print(signature.decode())


def verify(hash, signature):
      kp = Keypair.from_public_key("GDMG5Z4XL3CNGHK2GJD5TFIDRWRCBFVFV3WAFWFSBONWB6AKDODILHFZ")
 
      # Verify signature
      try:
            kp.verifying_key.verify(sig, msg, encoding='hex')
            print("Signature valid")
      except:
            print("Signature invalid")

File: examples/go/signature.go
package signature

import (
	"crypto/sha256"
	"encoding/hex"
	"fmt"

	"github.com/stellar/go/keypair"
)

func hash() {
	mesg := "Some message"

	// Hash
	hash := sha256.New()
	hash.Write([]byte(mesg))

	fmt.Println(hex.EncodeToString(hash.Sum(nil)))
}

func sign(hash string) {
	// GDMG5Z4XL3CNGHK2GJD5TFIDRWRCBFVFV3WAFWFSBONWB6AKDODILHFZ
	kp := keypair.MustParse("SDWMABEXMMUVENWEB73FB3EQJB5QSKOYIBXDOXAE6A3NIHIYRUQJSWXY")

	mesg, _ := hex.DecodeString(hash)

	// Sign
	sig, _ := kp.Sign(mesg)

	fmt.Println(hex.EncodeToString(sig))
}

func verify(hash, signature string) {
	kp := keypair.MustParse("GDMG5Z4XL3CNGHK2GJD5TFIDRWRCBFVFV3WAFWFSBONWB6AKDODILHFZ")

	// Decode hash and message
	mesg, _ := hex.DecodeString(hash)
	sig, _ := hex.DecodeString(signature)

	// Verify signature
	err = kp.Verify(mesg, sig)
	if err != nil {
		fmt.Println("Signature invalid")
	} else {
		fmt.Println("Signature valid")
	}
}
